include/linux/uaccess.h

```diff
@@ -74,18 +75,28 @@ static inline void force_uaccess_end(mm_segment_t oldfs)
 static __always_inline __must_check unsigned long
 __copy_from_user_inatomic(void *to, const void __user *from, unsigned long n)
 {
+	unsigned long res;
 	instrument_copy_from_user(to, from, n);
 	check_object_size(to, n, false);
-	return raw_copy_from_user(to, from, n);
+	res = raw_copy_from_user(to, from, n);
+#ifdef CONFIG_DF_DETECTION
```

Comment:
Kernel maintainers do not like excessive #ifdefs.
For our tools we usually do the following: declare the function in the header if the tool is enabled, and define it to an empty function if the tool is disabled. This way you can just call your function without any checks, and it will be simply optimized away if unneeded.
I.e.:
```
...
#ifdef CONFIG_DF_DETECTION
...
void dfetch_add_address(const void *addr, size_t len, unsigned long caller, void * kernel_addr);
...
#else
...
static inline void dfetch_add_address(const void *addr, size_t len, unsigned long caller, void * kernel_addr) {}
...
#endif
```
(check e.g. include/linux/kasan.h for other examples).